DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs

```diff
@@ -0,0 +1,144 @@
+﻿using DotNetFloripa.DataCore.DTO;
+using DotNetFloripa.ModelCore;
+using DotNetFloripa.ModelCore.Interfaces;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
+namespace DotNetFloripa.Data
+{
+    public class ExternalRepository : IAppRepository
+    {
+        public ExternalRepository()
+        {
+        }
+
+        private readonly TimeSpan ReloadThreshold = new TimeSpan(24, 0, 0);
+        private DateTime NextMeetupsLoad;
+        private DateTime NextCompaniesLoad;
+        private DateTime NextJobsLoad;
+
+        private bool ShouldLoadMeetups() => Events == null || Events.Length == 0 || NextMeetupsLoad < DateTime.Now;
+        private bool ShouldLoadCompanies() => Companies == null || Companies.Length == 0 || NextCompaniesLoad < DateTime.Now;
+        private bool ShouldLoadJobs() => Jobs == null || Jobs.Length == 0 || NextJobsLoad < DateTime.Now;
+
+        private T RequestJsonAs<T>(string baseURI, string resourceURI)
+        {
+            string result = string.Empty;
+
+            try
+            {
+                var httpClient = new HttpClient();
+                httpClient.BaseAddress = new Uri(baseURI);
+                httpClient.DefaultRequestHeaders.Accept.Clear();
+                httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                HttpResponseMessage response = httpClient.GetAsync(resourceURI).Result;
+
+                if (response.IsSuccessStatusCode)
+                    result = response.Content.ReadAsStringAsync().Result;
+            }
+            catch (Exception)
```

Comment:
Mas daí vai dar pau no site se por algum motivo o arquivo não for lido. É isso mesmo?